### PR TITLE
2D and 3D GLL points

### PR DESCRIPTION
**Pull request by adeebkor**
GLL points and GLL quadrature for quadrilateral and hexahedral element.

Add the option to compute the integration of quadrilateral and hexahedral Lagrangian element using GLL quadrature with GLL points.

### CODE REVIEW

**Review by mscroggs — APPROVED** [review]

**Comment by chrisrichardson** [comment]
Can you write a description. I am not sure what this PR does?
Also, is the GLL quadrature on triangles and tets? I thought it should onky work on quad/hex?

File: cpp/core/quadrature.cpp [hunk 1]
```diff
@@ -533,12 +589,42 @@ quadrature::compute_gauss_jacobi_rule(double a, int m)
 }
 //-----------------------------------------------------------------------------
 std::pair<Eigen::ArrayXd, Eigen::ArrayXd>
+quadrature::compute_gll_rule(double a, int m)
```
Comment (chrisrichardson) [inline cpp/core/quadrature.cpp:592]: What is "a"?

File: cpp/core/quadrature.h [hunk 2]
```diff
@@ -5,12 +5,11 @@
...
 #include "cell.h"
-#include <Eigen/Core>
+#include <Eigen/Dense>
 #include <utility>
 
-/// Integration using Gauss-Jacobi quadrature on simplices. Other shapes
-/// can be obtained by using a product.
-/// @todo - pyramid
+/// basix
+
```
Comment (chrisrichardson) [inline cpp/core/quadrature.h:12]: Don't delete documentation

File: cpp/core/quadrature.cpp [hunk 3]
```diff
@@ -197,6 +196,65 @@ make_gauss_jacobi_quadrature(cell::type celltype, int m)
...
+  }
+  case cell::type::prism:
+  {
+    throw std::runtime_error("Prism not yet supported");
+  }
+  case cell::type::pyramid:
+    throw std::runtime_error("Pyramid not yet supported");
+  case cell::type::triangle:
+    return quadrature::make_quadrature_triangle_collapsed(m);
+  case cell::type::tetrahedron:
```
Comment (chrisrichardson) [inline cpp/core/quadrature.cpp:250]: Is rhis correct?

**Review by chrisrichardson — CHANGES_REQUESTED** [review]

File: cpp/core/quadrature.cpp (hunk 1 above)
Comment (adeebkor) [reply cpp/core/quadrature.cpp:592]: I was using the `compute_gauss_jacobi_rule` as an example and forgot to remove "a". I'll remove in them now.

**Review by chrisrichardson — APPROVED** [review]

**Pull request merged**